backend/EkoFunkcje/EkoFunkcje/Features/Interventions/GetInterventionsInArea.cs

```diff
@@ -36,25 +36,42 @@ public GetInterventionsInArea()
         [FunctionName("GetInterventionsInArea")]
         public async Task<IActionResult> Run(
             [HttpTrigger(AuthorizationLevel.Function, "get", Route = "interventions/{latitude}/{longitude}")]
-            [RequestBodyType(typeof(AreaInterventionsFilterRequest), "AreaInterventionsFilterRequest")]AreaInterventionsFilterRequest filter,
+            [RequestBodyType(typeof(AreaInterventionsFilterRequest), "AreaInterventionsFilterRequest")]AreaInterventionsFilterRequest areaFilter,
             [Table(Config.InterventionsTableName, Connection = Config.StorageConnectionName)] CloudTable interventionsTable,
             string latitude, string longitude, ILogger log)
         {
-            var geoHash = GeoHasher.GetGeoHash(latitude + filter.GeoLatDiff, longitude + filter.GeoLngDiff);
+            string filter = GetFilter(areaFilter, latitude, longitude);
+            var interventions = await GetFilteredInterventions(areaFilter.Statuses, interventionsTable, filter);
+            return new JsonResult(interventions);
+        }
+
+        private async Task<IEnumerable<InterventionListItemResponse>> GetFilteredInterventions(
+            List<InterventionStatus> requestStatuses, CloudTable interventionsTable, string filter
+        ) {
             TableContinuationToken token = null;
 
-            var entities = new List<InterventionListItemResponse>();
+            var interventions = new List<InterventionListItemResponse>();
             do
             {
-                var queryResult = await interventionsTable.ExecuteQuerySegmentedAsync(new TableQuery<InterventionEntity>().Where(
-                    TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.Equal, geoHash)), token);
-                entities.AddRange(queryResult.Results.Select(x => _mapper.Map<InterventionListItemResponse>(x)));
+                var queryResult = await interventionsTable.ExecuteQuerySegmentedAsync(new TableQuery<InterventionEntity>().Where(filter), token);
+                interventions.AddRange(queryResult.Results.Select(x => _mapper.Map<InterventionListItemResponse>(x)));
                 token = queryResult.ContinuationToken;
             } while (token != null);
 
-            var filteredEntities = entities.Where(x => filter.Statuses.Contains(x.Status));
+            return  interventions.Where(intervention => requestStatuses.Contains(intervention.Status));
+        }
 
-            return new JsonResult(filteredEntities);
+        private static string GetFilter(AreaInterventionsFilterRequest areaFilter, string latitude, string longitude)
+        {
+            bool shouldUseGeoHashFilter =   Math.Abs(areaFilter.GeoLatDiff) < 0.001 && 
+                                            Math.Abs(areaFilter.GeoLngDiff) < 0.001;
+            if (shouldUseGeoHashFilter)
+            {
+                string geoHash = GeoHasher.GetGeoHash(latitude.Replace(',', '.'), longitude.Replace(',', '.'));
```

Comment:
Nie da się tego rozegrać jakimś Culture?

Reply:
Jako, że jesteśmy z Łukaszem współautorami kodu, pozwoliłem sobie sprawdzić i wychodzi na to, że InvariantCulture jest użyta wewnątrz metody GetGeoHash - usunąłem więc replace.